sklearn/feature_extraction/text.py

```diff
@@ -1496,8 +1496,10 @@ def transform(self, X, copy=True):
                 raise ValueError("Input has n_features=%d while the model"
                                  " has been trained with n_features=%d" % (
                                      n_features, expected_n_features))
-            # *= doesn't work
-            X = X * self._idf_diag
+
+            if copy:
+                X = X.copy()
+            X.data *= self._idf_diag.data[X.indices]
```

Comment:
It could be nice to have a small comment

```suggestion
            # sparse matrix does not support broadcasting but
            # with CSR matrix we can safely pick-up the indices.
            X.data *= self._idf_diag.data[X.indices]
```